### PR TITLE
Smart round

### PR DESCRIPTION
**Pull request by gka**
this PR adds 3 new functions to `shared` (which we already had in our old utils codebase)

* smartRound
* round
* significantDimension

see [README](https://github.com/datawrapper/shared/blob/9bd6afe8223cee992e45968739b5180e5d16e2d2/README.md#significantdimensionvalues-tolerance--number) to learn what the functions do.

### CODE REVIEW

File: round.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * round(1.2345, 2); // 1.23
+ * round(12345, -2); // 12300
+ *
+ * @param {number} value - the value to be rounded
+ * @param {number} decimals - the number of decimals
+ * @returns {string} - numeral.js compatible format string
+ */
+export default function round(value, decimals = 0) {
+    const base = Math.pow(10, decimals);
+    return Math.round(value * base) / base;
```
Comment (sto3psl) [inline round.js]: This returns a number but in the JSDoc comment you say it should return a numeral compatible `string`. Which one is correct?

File: round.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import test from 'ava';
+import round from './round';
+
+test('round to zero decimals', t => {
+    t.deepEqual(round(1.2345), 1);
+    t.deepEqual(round(1.2345, 0), 1);
```
Comment (sto3psl) [inline round.test.js]: `t.is` may have better error messages. I am curious, why are you using `t.deepEqual`?

File: significantDimension.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * without loosing information
+ *
+ * @exports significantDimension
+ * @kind function
+ *
+ * @example
+ * import {significantDimension} from '@datawrapper/shared/significantDimension';
+ * significantDimension([0,10,20,30]); // -1
+ *
+ * @param {array} values - list of input numbers
```
Comment (sto3psl) [inline significantDimension.js]: If values is an array of numbers you can also use this:
```suggestion
 * @param {number[]} values - list of input numbers
```

File: significantDimension.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    let check, diff;
+
+    const accepted = Math.floor(totalUniq * (1 - tolerance));
+
+    if (uniqValues.length < 3) {
+        // special case if there are only 2 unique values
+        return Math.round(
+            uniqValues.reduce(function(acc, cur) {
+                if (!cur) return acc;
+                var exp = Math.log(Math.abs(cur)) / Math.LN10;
```
Comment (sto3psl) [inline significantDimension.js]:
```suggestion
                const exp = Math.log(Math.abs(cur)) / Math.LN10;
```

File: significantDimension.test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        ],
+        dim: -3
+    },
+    { values: [50, 70, 90, 100, 130], dim: -1 },
+    { values: [0.01, 0.02, 0.03, 0.04, 0.04], dim: 2 }
+];
+
+tests.forEach((testCase, i) => {
+    test(`case ${i + 1}: check significant dimension`, t => {
+        t.deepEqual(significantDimension(testCase.values), testCase.dim);
```
Comment (sto3psl) [inline significantDimension.test.js]: Would use `deepEqual` only to actually compare deeply nested objects. I think ava recommends using `t.is` for most cases.

File: round.test.js (hunk 2 above)
Comment (gka) [reply round.test.js]: copy paste! :)

File: round.js (hunk 1 above)
Comment (gka) [reply round.js]: fixed the jsdoc

File: round.test.js (hunk 2 above)
Comment (gka) [reply round.test.js]: fixed

**Pull request merged**